### PR TITLE
feat(dropdown): focus on active item when opened

### PR DESCRIPTION
**Pull request by jcfranco**
Related issue: #396 

This adds the following enhancements to the dropdown: 

* Focuses the first active item or first one if there's none.
* Enables scrolling for long list of items.
* Fixes item focusing (it wasn't working before).
* Adds utility to help focus components or native elements.
* Fixes some tests and typings.
* Minor cleanup.

cc @dhrumil83

### CODE REVIEW

File: src/components/calcite-dropdown/calcite-dropdown.tsx [hunk 1]
```diff
@@ -257,24 +265,30 @@ export class CalciteDropdown {
...
   }
 
-  private openCalciteDropdown(e) {
+  private openCalciteDropdown() {
     this.active = !this.active;
-    // if invoked by key, focus item, and accomodate animation time
-    if (!e.detail && e.type !== "mouseenter") {
-      setTimeout(() => this.focusFirstItem(), 50);
-    }
+    const animationDelayInMs = 50;
```
Comment (paulcpederson) [inline src/components/calcite-dropdown/calcite-dropdown.tsx]: is there a reason you pulled this out into a variable?

**Comment by paulcpederson** [comment]
Nice, looks good

File: src/components/calcite-dropdown/calcite-dropdown.tsx (hunk 1 above)
Comment (jcfranco) [reply src/components/calcite-dropdown/calcite-dropdown.tsx]: Yes, to get rid of the comment. This can help avoid stale comments when refactoring.

Comment (paulcpederson) [reply src/components/calcite-dropdown/calcite-dropdown.tsx]: botcha

**Comment by macandcheese** [comment]
Awesome, great improvement! I do notice a flash-in of the scrollbar: https://jmp.sh/8sUymD9

Other than that it all looks good to me 🚀

**Comment by jcfranco** [comment]
@macandcheese I think I've fixed the broken transition. Could you take another look? 👁🔁 ❓

**Pull request merged**